tools/sqlite3_api_wrapper/sqlite3_udf_api/include/udf_struct_sqlite3.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "duckdb/common/types.hpp"
+
+using namespace duckdb;
+
+struct sqlite3_value {
+    union MemValue {
+        double r;           /* Real value used when MEM_Real is set in flags */
+        int64_t i;          /* Integer value used when MEM_Int is set in flags */
+        int nZero;          /* Extra zero bytes when MEM_Zero and MEM_Blob set */
+        const char *zPType; /* Pointer type when MEM_Term|MEM_Subtype|MEM_Null */
+    } u;
+    // uint16_t flags;
+    PhysicalType type;
+};
+
+struct sqlite3_context {
+    sqlite3_value result;
```

Comment:
The list of functions that can operate on the sqlite3_context is [here](https://sqlite.org/c3ref/context.html). We will need to implement all of these, although some of them can be stubs for now (e.g. `sqlite3_aggregate_context` is a stub).

Of interest for the contents of the `sqlite3_context` are:
* [sqlite3_set/getauxdata](https://sqlite.org/c3ref/get_auxdata.html)
* [sqlite3_result_error](https://sqlite.org/c3ref/result_blob.html)
* [sqlite3_user_data](https://sqlite.org/c3ref/user_data.html)
* [sqlite3_context_db_handle](https://sqlite.org/c3ref/context_db_handle.html)

All of the necessary information to implement these functions should be stored within the context.